index.bs

```diff
@@ -1168,21 +1201,42 @@ partial interface MLGraphBuilder {
+              - *epsilon*: a {{float}} scalar. A small value added to the variance of the batch in order to prevent computational error due to divide-by-zero. When it's not specified, the default value is 0.00001.
               - *activation*: an {{MLActivation}}. The optional activation function that immediately follows the normalization operation.
 
     **Returns:** an {{MLOperand}}. The batch-normalized N-D tensor of the same shape as the input tensor.
-
-    When *input* is a 4-D tensor of the *"nchw"* or *"nhwc"* layout, *options.axis* should be set to 1 or 3 respectively. The axis value designates the feature or channel count dimension of the input tensor.
+</div>
+<div algorithm=batchnorm>
+    The {{MLGraphBuilder/batchNormalization()}} method steps are:
+    1. Let |input| be the first argument. To validate |input|, run these substeps:
+        1. If |input| is not an object that [=implements=] {{MLOperand}}, then throw a "{{TypeError}}" {{DOMException}} and abort these steps.
+    1. Let |mean| be the second argument, representing a vector with the moving mean values for |input|. To validate |mean|, run the following substeps:
+        1. If |mean| is not an object that [=implements=] {{MLOperand}}, then throw a "{{TypeError}}" {{DOMException}} and abort these steps.
+        1. If |mean|.{{MLOperand/[[descriptor]]}}.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}} is not equal with |input|.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}} from which the dimension represented by |options|.axis is removed, then  throw a "{{TypeError}}" {{DOMException}} and abort these steps.
+    1. Let |variance| be the third argument, representing the moving variance values of |input|.
+    1. Let |options| be the fourth argument. To validate |options|, run these substeps:
+        1. If |options|.axis does not [=map/exist=], let |options|."axis" be 1.
+        1. If |options|.axis is not a number between 0 and the rank of |input|, then throw a "{{TypeError}}" {{DOMException}} and abort these steps.
+        1. If |input| is a 4-D tensor of the *"nchw"* layout, set |options|.axis to 1.
+        1. If |input| is a 4-D tensor of the *"nhwc"* layout, set |options|.axis to 3.
+            <div class="note">
+                The axis value designates the feature or channel count dimension of the input tensor.
+            </div>
+    1. Let |result| be an {{MLOperand}} representing the results. It may use the same underlying data as |input|.
+    1. Issue a request to the underlying platform to initialize the batch normalization, given |result| to store the results and |options|. Wait for completion.
+        <div class="informalsteps">
+            1. If |options|.activation [=map/exists=], the implementation MAY use it to fuse it with the batch normalization.
+        </div>
```

Comment:
If we feel compelled to add this note, there may be a better way to express it. The current wording doesn't read well because it doesn't explain what "fuse" in this context means, and what is "it" that the statement refers to. For example, a more complete wording may be "At the graph execution time, the implementation may choose to combine the execution of the standard batch normalization operation with the optional activation function provided by *options.activation* to improve the overall execution performance of the combined operation." etc.

Reply:
I agree, but I think the use of "fuse" should be handled in a separate issue/PR, consistently over the whole spec.